packages/core/src/metadata/MetadataDiscovery.ts

```diff
@@ -1152,7 +1152,10 @@ export class MetadataDiscovery {
       return;
     }
 
-    prop.unsigned = (prop.primary || prop.unsigned) && this.isNumericProperty(prop) && this.platform.supportsUnsigned();
+    prop.unsigned = (prop.primary || prop.unsigned)
+      && prop.unsigned !== false  // Allow explicit opt-out of unsigned for primary keys
+      && this.isNumericProperty(prop)
+      && this.platform.supportsUnsigned();
```

Comment:
won't this work the same? if not, let's keep your version. 

```suggestion
    prop.unsigned ??= (prop.primary || prop.unsigned) && this.isNumericProperty(prop) && this.platform.supportsUnsigned();
```

i'd consider this as a bug, the option is public and it does not work when explicitly used

Reply:
ah, yeah that definitely works -- that's much cleaner, thank you!